WordPress/gutenberg
# Prevent a block and console error for `input[type=email]`

## Description
Fixes an issue where if an `input[type="email"]` is at the top of a block, there's a block and console error. Though this is fixed once you reload the page.

## How has this been tested?
Steps to reproduce:
1. Use Chrome or Firefox, any WordPress version, like `5.5.1`, and the `master` branch of Gutenberg
1. Create a new post
1. Copy this into the console:
```js
wp.blocks.registerBlockType(
	'example/email-input-reproduce-issue',
	{
		title: 'Email Input',
		category: 'common',
		icon: 'wordpress-alt',
		keywords: [ 'Foo' ],
		attributes: {
			'email': {
				type: 'string',
			},
		},
		edit: function( props ) {
			return wp.element.createElement( 'input', { type: 'email', value: props.email } );
		},
	}
);
```
4. Insert the 'Email Input' block you just registered:<img width="554" alt="insert-block" src="https://user-images.githubusercontent.com/4063887/95902116-f181d400-0d59-11eb-91b0-61af9a1c6a7d.png">
5. Expected: The block should display
6. Actual: There's a block and console error:
<img width="818" alt="block-herb" src="https://user-images.githubusercontent.com/4063887/95902340-4de4f380-0d5a-11eb-9734-1e1afca82e98.png">

```
DOMException: Failed to set the 'selectionStart' property on 'HTMLInputElement': The input element's type ('email') does not support selection.
    at placeCaretAtHorizontalEdge (https://path/to/wp-content/plugins/gutenberg/build/dom/index.js?ver=cbf62a0bae5cfd2eeff038ee115edc0a:390:32)
```

That error is from:
https://github.com/WordPress/gutenberg/blob/efc836a778dc8ee74d94e1cb8eecef2d0878f612/packages/dom/src/dom.js#L281

It [looks like](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement) the `selectionStart` property applies 'only to text/number-containing or elements'. 

## Screenshots 
With this PR, there's no error:

<img width="773" alt="no-error" src="https://user-images.githubusercontent.com/4063887/95904587-525edb80-0d5d-11eb-9abe-ae99d9ecd9b8.png">


## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
